### ISSUE TITLE
Create Use Cases for Crop Rotation

### ISSUE THREAD

**Issue by hatchla** [ElektraInitiative/PermaplanT#1250]
### Tasks

 - [ ] Neighbours not existing simultaneously (temporal Relation)
 - [ ] Integration into Heatmap
 - [ ] 
 - [ ] 
 - [ ] 
 - [ ]

**Comment by markus2330** [comment]
Thx for writing the issue.

I added "Integration into Relation Layer".

**Comment by hatchla** [comment]
The [use case for the relation layer](https://github.com/ElektraInitiative/PermaplanT/blob/43b6ee52dcd0988a2f56fb9a0693ddd4a7872c5b/doc/usecases/current/relation_layer.md) already mentions the crop rotation... how should this be extended by an own use case?

**Comment by hatchla** [comment]
@markus2330 Could you state how these three usecases (as we talked about) are different than what is already mentioned in the corresponding usecases of e.g. heatmap?

**Comment by markus2330** [comment]
> Is the first use case here (temporal relation) already covered by [Suggest Plants](https://github.com/ElektraInitiative/PermaplanT/blob/1b4c349f57787f4acddda37cf9c818c12f2dfc8b/doc/usecases/assigned/suggest_plants.md) or how is it different for you?

Yes, it is different, temporal relation is part of heatmap and relations. The difference is:

1. crop rotation use case are recommendations of spots on the map for a specific plant for a specific time
2. suggest plants is about recommending various plants on a specific spot on the map for a specific time

> how should this be extended by an own use case?

The relation layer specific part can be part of the relation layer use case. It is only that then the relation layer and heatmap usecases are huge.

> Could you state how these three usecases (as we talked about) are different than what is already mentioned in the corresponding usecases of e.g. heatmap?

The heatmap does not even mention crop rotation currently.

One way to solve this elegantly without duplication:

1. a short heatmap and relation layer use case, with a bunch of links to other use cases, including crop_rotation
2. these other use cases then explain what they do for the user in the case of heatmap and relation

**Comment by hatchla** [comment]
We may need to talk about this in a meeting. Generally and also in our cases, use cases should not be one single scrum issue but rather multiple ones that are as small as somehow possible and therefore I think its unproblematic to make the usecases big.
And if we want to make the use cases clearer, easier to read and therefore smaller/splitting them up, we should do that consequently and not only with a new feature like this...

**Comment by hatchla** [comment]
> One way to solve this elegantly without duplication:
>
> 1. a short heatmap and relation layer use case, with a bunch of links to other use cases, including crop_rotation
> 2. these other use cases then explain what they do for the user in the case of heatmap and relation

This approach seems reasonable to me for this specific use cases, the earlier mentioned aspects should still be considered.
